### PR TITLE
[WIP][RHCLOUD-18283] Endpoint tests

### PR DESCRIPTION
**Pull request by petracihalova**
This PR is about adding missing test for `EndpointListAuthetications()` and about fixing related code with main goal to have tests that pass when secret store is database and vault.
- special fixtures for `TestEndpointDelete()` added (previous test manipulate with test data which I needed for different test)
- `TestAuthenticationEdit()` updated (the test failed with secret store = vault)
- test for `EndpointListAuthentications() `created (first version only for secret store = database)
- fixed logic around start and end index in keys array (related with limit, offset) in `List()` method in `authentication_dao.go`
- updated `Read()` and `List() `method in mock vault and added new auths fixtures - so skip condition in several tests was removed because now these tests are working for secret store = vault
- `authFromVault()` from `authentication_dao.go` updated = added parsing tenant id from secret (needed for `TestAuthenticationEdit()` tests - this test started failing again with secret store = vault)
- in the end created missing bad request and not found missing tests for `EndpointListAuthentications()`

**JIRA:** [RHCLOUD-18283](https://issues.redhat.com/browse/RHCLOUD-18283)

### CODE REVIEW

**Comment by petracihalova** [comment]
rebased & conflicts fixed

File: internal/testutils/mocks/mock_vault.go [hunk 1]
```diff
@@ -15,24 +16,65 @@ type MockVault struct {
 
 var VaultPath = []string{fmt.Sprintf("Application_%d_%v", fixtures.TestTenantData[0].Id, fixtures.TestAuthenticationData[0].ID)}
 
-func (m *MockVault) Read(path string) (*api.Secret, error) {
-	if path != fmt.Sprintf("secret/data/%d/%v", fixtures.TestTenantData[0].Id, VaultPath[0]) {
-		return nil, nil
+// getVaultPathsFromFixtures creates list of fault paths
```
Comment (MikelAlejoBR) [inline internal/testutils/mocks/mock_vault.go]: "fault" -> "vault"

**Review by MikelAlejoBR — APPROVED** [review]
LGTM!

File: internal/testutils/mocks/mock_vault.go (hunk 1 above)
Comment (petracihalova) [reply internal/testutils/mocks/mock_vault.go]: really good eye 👁

**Comment by petracihalova** [comment]
I decided to split these changes in several PRs to have it more well-arranged. Simply I will separate the parts related with new tests for EndpointListAuthetications() and other changes.

**Comment by petracihalova** [comment]
will be split into several new PRs (or reopened when needed)

**Pull request closed without merging**